src/Microsoft.DotNet.Darc/DarcLib/Helpers/NullSafeUnixPathDictionary.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+using System.Collections.Generic;
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib.Helpers;
+
+/// <summary>
+/// A dictionary that treats null UnixPath keys as equivalent to UnixPath(".") for root directory operations.
+/// This allows using null to represent the root directory while maintaining dictionary functionality.
+/// </summary>
+/// <typeparam name="TValue">The type of values in the dictionary</typeparam>
+public class NullSafeUnixPathDictionary<TValue> : IEnumerable<KeyValuePair<UnixPath?, TValue>>
```

Comment:
This is not a good solution I think. It should be dealt with on a lower level - you should just overload the `UnixPath.Equals()` (and also `GetHash..`) so that `null` equals `""` equals `.`. Then all other things like dictionaries start behaving appropriately and you'd be safe against people using Linq or just a different structure than dictionary.

I'd also introduce a static `UnixPath.Empty` or similar that could be referenced same way `string.Empty` is used.

I'd not put `/` here as that has a bit different meaning.

Reply:
I implemented this. I guess it has a small downside that it is possible that someone uses UnixPath in a dictionary, and doesn't know about the overrides in this PR and gets some weird behavior. But that's probably unlikely enough

Reply:
What is nice is the fact that:
- `foo` / `""` / `bar`
- `foo/` / `.` / `bar`
- `foo` / `null` / `bar`

means the same when talking about paths so it does not matter